metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -2040,84 +2042,13 @@ class MetalsLanguageServer(
           languageClient.showMessage(Messages.Worksheets.unableToExport)
           Future.successful(()).asJavaObject
         }
-
-      case ServerCommands.InsertInferredType(textDocumentParams) =>
-        CancelTokens.future { token =>
-          val uri = textDocumentParams.getTextDocument().getUri()
-          for {
-            edits <- compilers.insertInferredType(textDocumentParams, token)
-            _ = logging.logErrorWhen(
-              edits.isEmpty(),
-              s"Could not infer type at ${textDocumentParams.getPosition()} in file\n${buffers
-                  .get(uri.toAbsolutePath)}",
-            )
-            workspaceEdit = new l.WorkspaceEdit(Map(uri -> edits).asJava)
-            _ <- languageClient
-              .applyEdit(new ApplyWorkspaceEditParams(workspaceEdit))
-              .asScala
-          } yield ().asInstanceOf[Object]
-        }
-      case ServerCommands.ExtractMethod(
-            ServerCommands.ExtractMethodParams(doc, range, extractionPos)
+      case actionCommand
+          if codeActionProvider.allActionCommands.exists(
```

Comment:
Wdyt od having Set with all code actions ids?

Reply:
Addeded for both ServerCommands and code action commands.